include/swift/SIL/SILParser.h

```diff
@@ -0,0 +1,67 @@
+//===--- SILParser.h - SIL Undef Value Representation -----------*- C++ -*-===//
+//
+// This source file is part of the Swift.org open source project
+//
+// Copyright (c) 2014 - 2019 Apple Inc. and the Swift project authors
+// Licensed under Apache License v2.0 with Runtime Library Exception
+//
+// See https://swift.org/LICENSE.txt for license information
+// See https://swift.org/CONTRIBUTORS.txt for the list of Swift project authors
+//
+//===----------------------------------------------------------------------===//
+
+#ifndef SWIFT_SIL_PARSER_H
+#define SWIFT_SIL_PARSER_H
+
+#include "swift/AST/DeclContext.h"
+#include "swift/Basic/LLVM.h"
+#include "swift/Parse/Parser.h"
+#include "swift/SIL/SILInstruction.h"
+
+namespace swift {
+namespace syntax {
+
+//===----------------------------------------------------------------------===//
+// SILParser local value container
+//===----------------------------------------------------------------------===//
+
+struct UnresolvedValueName {
+  StringRef Name;
+  SourceLoc NameLoc;
+
+  bool isUndef() const { return Name == "undef"; }
+};
+
+class SILParsedLocalValueMap {
+  SILModule &module;
+  bool &hadError;
```

Comment:
I really don't like that we have to have this here. I'd like if this class can be as close to a map as possible. It shouldn't worry about diagnostics and errors. Anyway, that's a change for later.

Reply:
Make this a callback so the interfaces are clean.

Reply:
Please make this a private header in ./lib/SIL/Parser. This doesn't need to be exposed outside the library.

Reply:
Good call. Made it a callback. I, unfortunately, had to expose the member and set it after initializing `SILParsedLocalValueMap` because you can't capture a member in a lambda in that class' constructor. 

What do you mean by a private header? Is that a CMake option?

Reply:
A private header is something in ./lib that isn't exposed. I would create a private .h file in ./lib/SIL/Parser. Maybe call it ./lib/SIL/Parser/SILParsedLocalValueMap.h

Reply:
Good to know. Will do.

Reply:
Done.